snowfakery/data_generator_runtime.py

```diff
@@ -131,12 +172,21 @@ class Globals(yaml.YAMLObject):
     yaml_dumper = yaml.SafeDumper
     yaml_tag = "!snowfakery_globals"
 
-    def __init__(self, start_ids: Optional[Dict[str, id]] = None, today: date = None):
+    def __init__(
+        self,
+        start_ids: Optional[Dict[str, id]] = None,
+        today: date = None,
+        name_slots: Mapping[str, str] = (),
```

Comment:
A mapping type that defaults to a tuple?

Reply:
It's immutable and it responds to iter, __contains__. There is no "empty immutable mapping type." It actually works nicely although its true that mypy doesn't like it. Slightly frustrating because it is more mapping-like than None, which mypy does allow. (of course None has all sorts of special processing around it)

I guess I'll avoid that convention in the future to minimize future rework when we use MyPy.

Reply:
https://github.com/SFDO-Tooling/Snowfakery/commit/7b57423dafe18b0e5d76f02ec3d3eb453f4a4ed2#diff-643cb5ba59a4a93cb618c80e5ce3fc18R188